### PR TITLE
For Dynamo storage, throw an exception when a Put set a secondary index column value to null

### PR DESCRIPTION
**Pull request by Torch3333**
Please have a look.

### CODE REVIEW

File: core/src/main/java/com/scalar/db/storage/common/checker/DynamoOperationChecker.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package com.scalar.db.storage.common.checker;
+
+import com.scalar.db.api.Put;
+import com.scalar.db.api.TableMetadata;
+import com.scalar.db.common.TableMetadataManager;
+
+public class DynamoOperationChecker extends OperationChecker {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/storage/common/checker/DynamoOperationChecker.java:7]: I think we should put `DynamoOperationChecker` in the `com.scalar.db.storage.dynamo` package. What do you think?

**Review by brfrn169 — COMMENTED** [review]
Thank you! LGTM overall. I left a comment. Please take a look when you have time!

File: core/src/main/java/com/scalar/db/storage/common/checker/DynamoOperationChecker.java (hunk 1 above)
Comment (Torch3333) [reply core/src/main/java/com/scalar/db/storage/common/checker/DynamoOperationChecker.java:7]: Yes, that makes more sense. Thank you

**Review by brfrn169 — APPROVED** [review]
LGTM! Thank you!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**